ecs-metadata-collector/src/main/java/com/emc/ecs/metadata/bo/BillingBO.java

```diff
@@ -267,6 +268,215 @@ public void getObjectBucketData( Map<NamespaceBucketKey, ObjectBucket> objectBuc
 		
 	}
 	
+	/**
+	 * 
+	 * @param objectPerNamespaceMap
+	 * @param objectBucketMap
+	 * @param collectionTime
+	 * @param billDAO
+	 */
+	private void collectObjectBucketDataByBucket(Map<String, Set<ObjectBucket>> objectPerNamespaceMap,
+			Map<NamespaceBucketKey, ObjectBucket> objectBucketMap, Date collectionTime, BillingDAO billDAO,
+			String namespace, String bucketname) {
+		long objCounter = 0;
+		BucketRequest bucketRequest = new BucketRequest();
+		bucketRequest.setName(bucketname);
+		bucketRequest.setNamespace(namespace);
+		ObjectBuckets objectBucketsResponse = client.getBucketInfo(bucketRequest);
+
+		if (objectBucketsResponse == null) {
+			return;
+		}
+
+		logger.info("Collect Billing Data for namespace: " + namespace + " and bucket: "+ bucketname);
+		objCounter += (objectBucketsResponse.getObjectBucket() != null) ? objectBucketsResponse.getObjectBucket().size()
+				: 0;
+
+		// Push collected info into datastore
+		if (billDAO != null) {
+			billDAO.insert(objectBucketsResponse, collectionTime);
+		}
+
+		// Add to return map per namespace and bucket key
+		if (objectBucketsResponse.getObjectBucket() != null && objectBucketMap != null) {
+			for (ObjectBucket objectBucket : objectBucketsResponse.getObjectBucket()) {
+				//if (bucketname.equals(objectBucket.getName())) {
```

Comment:
I guess verifying the bucket name again is redundant. Should we remove the commented code so it is cleaner.